### ISSUE TITLE
Error on instance generation setting extensions to fixed values

### ISSUE THREAD

**Issue by markkramerus** [FHIR/sushi#113]
In instances, setting extensions seems to work in SUSHI, but something in the generated resource causes IG Publisher to crash. The offending lines seems to follow the same pattern, e.g.:

`* extension[TreatmentIntent].valueCodeableConcept = SCT#373808002 "Curative - procedure intent"`

`* extension[RadiationDose].extension[TotalRadiationDoseDelivered].valueQuantity = UCUM#cGy`

`* extension[USCoreBirthSex] = GENDER#F`

See examples.fsh for commented-out lines where this error occurs.

This error was "silent" in SUSHI, but only because apparent when the IG Publisher died -- hence, very hard to diagnose and pin down. This makes it a high priority to fix.

**Comment by ngfreiter** [comment]
I investigated this a little, and it looks like the problem is that we need automatically to set the URLs of non-inline defined extensions when creating the instance. For example, if the FSH causing this error is changed to be:
```
* extension[TreatmentIntent].valueCodeableConcept = SCT#373808002 "Curative - procedure intent"
* extension[TreatmentIntent].url = "http://hl7.org/fhir/us/mcode/StructureDefinition/TreatmentIntent"
* extension[RadiationDose].extension[TotalRadiationDoseDelivered].valueQuantity = UCUM#cGy 
* extension[RadiationDose].extension[TotalRadiationDoseDelivered].valueQuantity.value = 1200.0
* extension[RadiationDose].url = "http://hl7.org/fhir/us/mcode/StructureDefinition/RadiationDose"
* extension[RadiationDose].extension[TotalRadiationDoseDelivered].url = "http://hl7.org/fhir/us/mcode/StructureDefinition/TotalRadiationDoseDelivered"
```
Then publishing should successfully complete. However, the lines in the above FSH where the URL is set should happen automatically, we shouldn't have to explicitly set the URL, because the URL can be referred from the the slice on the Structure Definition we are making an instance of. This doesn't appear to be an issue when the extension is defined inline. It only seems to be an issue when the extension is separately defined with the `Extension` keyword, and is then referenced in a profile.

**Comment by ngfreiter** [comment]
Also note that `extension[USCoreBirthSex] = GENDER#F` will fail because of #108

**Comment by cmoesel** [comment]
Fixed in 0.6.1

**Comment by markkramerus** [comment]
confirmed fixed.
